service/storage_service.cc

```diff
@@ -169,6 +169,21 @@ void storage_service::register_metrics() {
     });
 }
 
+std::optional<gms::inet_address> get_owner_of_tokens(locator::mutable_token_metadata_ptr tmptr, const std::unordered_set<dht::token>& tokens) {
+    std::unordered_set<gms::inet_address> nodes;
+    for (auto& t : tokens) {
+        auto addr = tmptr->get_endpoint(t);
+        if (addr) {
+            nodes.insert(*addr);
+        }
```

Comment:
else return std::nullopt, so we only match if all of the tokens were accounted for.

Reply:
Yes, this is what is wanted. The tokens of the two nodes (replacing and node to be replaced) should be exactly the same. I mentioned somewhere else, we should check they are identical.